runtimes/runtimes/standalone.ts

```diff
@@ -181,6 +182,61 @@ export const standalone = (props: RuntimeProps) => {
                         })
                     }),
                 getFileSize: path => stat(path),
+                getServerDataFolder: serverName => {
+                    const initializeParams = lspRouter.clientInitializeParams
+                    const clientSpecifiedLocation = initializeParams?.initializationOptions?.aws.clientDataFolder
+                    if (clientSpecifiedLocation) {
+                        return path.join(clientSpecifiedLocation, serverName)
+                    }
+
+                    const clientFolderName = getClientNameFromParams(initializeParams)
+                    const standardizedClientFolderName = standardizeFileName(clientFolderName)
+
+                    const appDataFolder = getPlatformAppDataFolder()
+                    return appDataFolder
+                        ? path.join(appDataFolder, standardizedClientFolderName, serverName)
+                        : path.join(os.homedir(), `.${standardizedClientFolderName}`, serverName)
+
+                    function getPlatformAppDataFolder(): string {
+                        const platform = process.platform
+                        switch (platform) {
+                            case 'win32':
+                                return process.env.APPDATA || path.join(os.homedir(), 'AppData', 'Roaming')
+
+                            case 'darwin':
+                                return path.join(os.homedir(), 'Library', 'Application Support')
+
+                            case 'linux':
+                                return process.env.XDG_DATA_HOME || path.join(os.homedir(), '.local', 'share')
+
+                            default:
+                                return ''
+                        }
+                    }
+
+                    function getClientNameFromParams(initializeParams: InitializeParams | undefined): string {
+                        const clientInfo = initializeParams?.clientInfo
+                        const awsClientInfo = initializeParams?.initializationOptions?.aws.clientInfo
+
+                        return [
+                            awsClientInfo?.name || clientInfo?.name || '',
+                            awsClientInfo?.version || clientInfo?.version || '',
```

Comment:
Is there any expectation that this folder would be used to persist data across updates to the user's IDE and/or extension?  If this isn't expected to hold long-lived persisted values, should it be moved to a temp folder instead?

If this should be version specific:

1. How will data that should be persisted be migrated?
2. What will clean up the long tail of unused folders after updates?

Reply:
I agree, data should be persisted across updates. Also, are you sure we'll need to append both client and extension names? I think we'll end up with long path which risks to hit the limits.

Reply:
* Yes, you're right - I hadn't considered version updates. 
* We append both the client and extension names because many extensions could use our protocol, which might lead to collisions.
* I've reduced the folder name's max size a bit in the  `standardizeFolderName` function.

Reply:
"We append both the client and extension names because many extensions could use our protocol, which might lead to collisions." - Agreed, but I think in the latest implementation version you are using client or extension name now. Shouldn't it contain both?

Reply:
It does contain both, if the values are present.